Client/Frontend/Home/PanelDataObservers.swift

```diff
@@ -48,28 +49,33 @@ class ActivityStreamDataObserver: DataObserver {
         events.forEach { NotificationCenter.default.removeObserver(self, name: $0, object: nil) }
     }
     
-    func invalidate(highlights: Bool) {
+    func refresh(highlights: Bool) {
+        guard !profile.isShutdown else {
+            return
+        }
         self.delegate?.willInvalidateDataSources()
 
-        let notify = {
+        // Highlights are cached for 15 mins
+        let invalidateHighlights = highlights ? true : (Date.now() - lastInvalidation > invalidationTime)
```

Comment:
Is this something that will make a big difference. I dont mind if this isnt _always_ exactly right. Using uptime is annoying, having to handle the case where the device has just booted and the uptime is too small is no fun. We had a bunch of bugs in the Password checking because of this